addOns/commonlib/src/main/java/org/zaproxy/addon/commonlib/ResourceIdentificationUtils.java

```diff
@@ -0,0 +1,99 @@
+/*
+ * Zed Attack Proxy (ZAP) and its related class files.
+ *
+ * ZAP is an HTTP/HTTPS proxy for assessing web application security.
+ *
+ * Copyright 2021 The ZAP Development Team
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.zaproxy.addon.commonlib;
+
+import java.util.regex.Pattern;
+import org.parosproxy.paros.network.HttpMessage;
+
+public final class ResourceIdentificationUtils {
+    private ResourceIdentificationUtils() {}
+
+    private static final Pattern PATTERN_FONT_EXTENSIONS =
+            Pattern.compile("(?:\\.ttf|\\.woff|\\.woff2|\\.otf)\\z", Pattern.CASE_INSENSITIVE);
+    private static final Pattern PATTERN_JS_EXTENSIONS =
+            Pattern.compile("(?:\\.js|\\.jsx)\\z", Pattern.CASE_INSENSITIVE);
+
+    /**
+     * Returns whether or not the given {@link HttpMessage} has a font content type in its response
+     * or request headers.
```

Comment:
It's not checking the request headers.

Reply:
I think that’s based on my text, I was trying to generically explain bother are checked and figured url is part of the request header technically (prime header) 🤷‍♂️

Open to suggestions.

Reply:
The prime header is a Paros naming, it's called start line (request line for the request). I'd prefer be clear and say it's checking the URL for known file extensions.